--- /home/dmitry/libs/mongo/src/mongo/util/ramlog.cpp	2013-02-17 22:49:18.268603371 +0300
+++ /home/dmitry/dev/robomongo/src/robomongo/shell/util/ramlog.cpp	2013-02-17 22:09:31.108675734 +0300
@@ -15,11 +15,19 @@
  *    limitations under the License.
  */
 
+#ifdef ROBOMONGO
+#include "mongo/pch.h"
+#include "mongo/util/log.h"
+#include "mongo/util/ramlog.h"
+#include "mongo/util/mongoutils/html.h"
+#include "mongo/util/mongoutils/str.h"
+#else
 #include "pch.h"
 #include "log.h"
 #include "ramlog.h"
 #include "mongoutils/html.h"
 #include "mongoutils/str.h"
+#endif
 
 namespace mongo {
 